modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/search/bar/portlet/SearchBarPortlet.java

```diff
@@ -85,13 +89,37 @@ public void render(
 			new SearchBarPortletPreferencesImpl(
 				Optional.ofNullable(renderRequest.getPreferences()));
 
-		PortletSharedSearchResponse portletSharedSearchResponse =
-			portletSharedSearchRequest.search(renderRequest);
+		SearchBarPortletDisplayContext searchBarPortletDisplayContext = null;
 
-		SearchBarPortletDisplayContext searchBarPortletDisplayContext =
-			_buildDisplayContext(
-				portletSharedSearchResponse, renderRequest,
-				searchBarPortletPreferences);
+		String destination = searchBarPortletPreferences.getDestinationString();
+
+		if (!Validator.isBlank(destination)) {
```

Comment:
@slnn Here you should not depend on `Validator.isBlank`. If destination is really empty, you should let it go with the normal path. The shortcut path should only be chosen when the destination is really unreachable.

Reply:
@slnn You should just move the private method from `SeachBarPortletDisplayContextBuilder` into `SearchBarPortlet` because after doing this we'll never need the build to do this check.

Reply:
Hi @dantewang 

When the destination is unreachable. the destination is not empty, from the logic in SearchBarPortletDisplayContextBuilder.build();  when not layout's friendlyurl is "/search", it means the destination is unreachable.

Reply:
Please see

if (Validator.isBlank(_destination)) {
			searchBarPortletDisplayContext.setSearchURL(_getURLCurrentPath());
		}
		else {
			String destinationURL = _getDestinationURL(_destination);

			if (destinationURL == null) {
				searchBarPortletDisplayContext.setDestinationUnreachable(true);
				searchBarPortletDisplayContext.setRenderNothing(true);
			}
			else {
				searchBarPortletDisplayContext.setSearchURL(destinationURL);
			}
		}

Reply:
@slnn Your current logic does nothing if the destination is empty. This is not what we want. We only want to check whether the destination is unreadable (which is not empty).

Reply:
OK, I see